### PR TITLE
feat(azure): keep azure provider to listen the app gateway event

### PR DESCRIPTION
**Pull request by FogDong**
**What this PR does / why we need it**:

feat(azure): keep azure provider to listen the app gateway event

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to caicloud/service#770

**Special notes for your reviewer**:

/cc @whalecold @hanxueluo 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/provider/providers/azure/azure.go [hunk 1]
```diff
@@ -195,10 +195,10 @@ func (f *azure) syncLoadBalancer(obj interface{}) error {
...
 	lb = nlb.DeepCopy()
 
-	if lb.Spec.Providers.Azure == nil {
-		// It is not my responsible, clean up legacies
-		return f.cleanup(lb, true)
-	}
+	// if lb.Spec.Providers.Azure == nil {
+	// 	// It is not my responsible, clean up legacies
+	// 	return f.cleanup(lb, true)
+	// }
```
Comment (whalecold) [inline pkg/provider/providers/azure/azure.go]: 只删除这一段的话会有问题的吧 如果这个 lb 既没有 azure 也没有 appgateway 也会被创建了。

Comment (FogDong) [reply pkg/provider/providers/azure/azure.go]: 有可能一开始没有 appgateway 后来开启了

Comment (whalecold) [reply pkg/provider/providers/azure/azure.go]: 后来开启的话后面还是会创建的。

**Pull request merged**